packages/expo-calendar/ios/Dialogs/PromiseRef.swift

```diff
@@ -0,0 +1,26 @@
+import ExpoModulesCore
+
+// TODO vonovak: this could / should be provided by Expo modules? Then it could be used in other modules as well.
```

Comment:
given how promises work in JS, it can only be fulfilled once, question is what happens when you try that multiple times from native. Previously it used to crash. I have merged `main` into this branch and I verified that resolving a promise multiple times from native doesn't lead to a crash as it did before. I have therefore removed the `PromiseRef.swift` class altogether because it doesn't serve any purpose now and the crash doesn't happen.

But the code will still, in edge cases, attempt to resolve the promise twice. The comment I took from the deleted code explains it:
// double resolve of the promise happens
// when `presentationControllerDidDismiss` is called together with
// methods from EKEventEditViewDelegate / EKEventViewDelegate.
// That can be achieved on iPad if you e.g. quickly delete an event and then tap outside of the dialog to dismiss it.